src/main/java/net/snowflake/client/jdbc/SnowflakeDatabaseMetaData.java

```diff
@@ -3223,7 +3223,8 @@ private ResultSet executeAndReturnEmptyResultIfNotFound(
       resultSet = statement.executeQuery(sql);
     } catch (SnowflakeSQLException e) {
       if (e.getSQLState().equals(SqlState.NO_DATA)
-          || e.getSQLState().equals(SqlState.BASE_TABLE_OR_VIEW_NOT_FOUND)) {
+          || e.getSQLState().equals(SqlState.BASE_TABLE_OR_VIEW_NOT_FOUND)
+          || e.getMessage().contains("Operation is not supported in reader account")) {
```

Comment:
Is there any ways other than leveraging message, e.g. sql state code, to recognize the situation?

Reply:
True, if there is a solution other than checking against a string message, that would be preferable because the messages get updated and change sometimes whereas the codes usually stay the same.

Reply:
The sql state returns SYNTAX_ERROR and error code from the server returns UNSUPPORTED_DB_OPERATION, so neither are specific to getProcedures not being supported on reader accounts.